### ISSUE TITLE
PVR_RECORDING iEpisode not what expected

### ISSUE THREAD

**Issue by fuzzard** [kodi-pvr/pvr.demo#59]
I've implemented strEpisodeName, iSeriesNumber, iEpisodeNumber for PVR_Recordings as per https://github.com/kodi-pvr/pvr.demo/pull/58

In conjunction with the skin changes to estuary (https://github.com/xbmc/xbmc/pull/13300), there is a display issue when a recording only has an iEpisodeNumber, but im not seeing anything obvious from the addon or skinning side, so not sure if its a kodi problem or not?

The issue can be seen when viewing Recording "Demo TV Recording entry 2"  and also deleted recording "Demo TV Recording entry 12" . Deleted recording  "Demo TV Recording entry 11" also is wrong, as it doesnt show season number.

From what i can see, they use the exact same include that is working without any issues for the Guide view and the Channel view, so i dont think its a skinning issue as such, as i would expect the problem to be there with any view using the include.

The ints are being set correctly into the PVRDemoRecording object (logging was used to verify this locally), so i dont think its the demo addon either thats mangling the episode/series number.

If you can point me to where kodi receives/handles the iepisode and passes it onto the skin, im happy to go on a hunt, ive just no idea how the skin receives the data after the entries have been transferred to kodi.

**Recordings**

This is correct
![screen shot 2018-01-05 at 2 30 23 pm](https://user-images.githubusercontent.com/2861319/34595606-23e1847a-f225-11e7-9d17-71fcfb68f37e.png)

This should show E3 Demo TV Recording 2 Episode Name
![screen shot 2018-01-05 at 2 30 25 pm](https://user-images.githubusercontent.com/2861319/34595610-2a931310-f225-11e7-853e-45cbcaca9e69.png)

This is correct
![screen shot 2018-01-05 at 2 30 26 pm](https://user-images.githubusercontent.com/2861319/34595621-3259687e-f225-11e7-9e2a-6cd68f0b24b7.png)

**Deleted Recordings**

This one should show S1E2 Demo TV Recording entry 11 Episode Name
![screen shot 2018-01-05 at 2 30 35 pm](https://user-images.githubusercontent.com/2861319/34595629-3f3972b4-f225-11e7-8813-2870a6f17f80.png)

This should show E3 Demo TV Recording entry 12 Episode Name
![screen shot 2018-01-05 at 2 30 37 pm](https://user-images.githubusercontent.com/2861319/34595633-419323c0-f225-11e7-80b5-a0e241a72890.png)

Episode title only is correct
![screen shot 2018-01-05 at 2 30 43 pm](https://user-images.githubusercontent.com/2861319/34595637-43b7ea5a-f225-11e7-9de4-38cc5156dc1d.png)

**Comment by fuzzard** [comment]
Another screen shot to show the issue after a did up a commit to use the season: X episode: y format da-anda hoped for in the skin PR

The S3 is the issue. I dunno where the S is coming from at all, but the skin is populating it from $INFO[ListItem.Episode, xxx] call in the skin.
![screen shot 2018-01-05 at 4 02 48 pm](https://user-images.githubusercontent.com/2861319/34597399-1567cf0a-f232-11e7-9971-dde5339be34c.png)
![screen shot 2018-01-05 at 4 02 57 pm](https://user-images.githubusercontent.com/2861319/34597401-16f79ac6-f232-11e7-870e-d929050e409a.png)

**Comment by ksooo** [comment]
As "E" and "S" are swapped sometimes I'm pretty sure it's related to the skin. The skin defines if "S" and "E" appear and in which order.

> Another screen shot to show the issue after a did up a commit to use the season: X episode: y format da-anda hoped for in the skin PR

 Do not do this change. I will not approve it. I'm the PVR maintainer and my word is the last on such things. ;-)

**Comment by ksooo** [comment]
Wait, could be a core issue. I will take a look later today.

**Comment by fuzzard** [comment]
Yeah no problems, wont push that commit. It looks ugly and bloats the screen info imo, but thought id demo it up just in case.

**Comment by ksooo** [comment]
> Wait, could be a core issue. I will take a look later today.

https://github.com/xbmc/xbmc/blob/master/xbmc/GUIInfoManager.cpp#L9913

I have no idea why an episode number should be prefixed with "S" if season is 0. The skin prepends an "E" and result is "ES1" (for episode 1). @Montellese ?

**Comment by fuzzard** [comment]
Looking at it, it seems the assumption is that if there is no season, then its a "special", hence the S apparently. My question is why this assumption isnt used for EPG or for channel? Should this assumption of a special be implemented into channel and guide? Ill let you guys discuss how you want to handle that. I just know EPG data coming from SiliconDust for a HDHomerun often has shows listed with only an episode number, and they arent specials.

On a side note, should the listitem for series and episode do a check for a PVRRecordingInfoTag like it does for an EPGInfoTag and ChannelInfoTag? Adding something like the below for LISTITEM_EPISODE for example?

`      else if (item->HasPVRRecordingInfoTag())
      {
        if (item->GetPVRRecordingInfoTag()->m_iSeason > 0)
          iSeason = item->GetPVRRecordingInfoTag()->m_iSeason;
        if (item->GetPVRRecordingInfoTag()->m_iEpisode > 0)
          iEpisode = item->GetPVRRecordingInfoTag()->m_iEpisode;
      }`

**Comment by fuzzard** [comment]
Actually, one of those if statements is succeeding for a recording, and because i set iSeries to 0 in the addon, iSeries is then being set to 0 from the -1 in that method. Then it goes into the if statement with the additional formatting with the S as iSeason is now == 0.

Solution from an addon perspective is to just set recording.iSeriesNumber = -1;. I would consider this a workaround that doesnt fix the actual issue.
I assume if we were to do a check for a PVRRecordingInfoTag, it would then set season to > 0, or leave at -1 like it does with the others. Would make the formatting for a special pointless (it is for channels and guide as it stands)

Your call on how you want to handle it. Happy to do a PR for which ever way you want to fix it, just let me know.

**Comment by FernetMenta** [comment]
AFAIK the letters S and E are required for scrapers, TVDB. Last time I tried it was broken for PVR recordings. If this is gonna touched maby look at the craping issue too.

**Comment by ksooo** [comment]
> On a side note, should the listitem for series and episode do a check for a PVRRecordingInfoTag like it does for an EPGInfoTag and ChannelInfoTag? Adding something like the below for LISTITEM_EPISODE for example?

 No. `CPVRRecording` is derived from `CVideoInfoTag` which is handled properly already.

**Comment by ksooo** [comment]
> Looking at it, it seems the assumption is that if there is no season, then its a "special", hence the S apparently. My question is why this assumption isnt used for EPG or for channel? Should this assumption of a special be implemented into channel and guide?

 Yes. It's missing there.

**Comment by ksooo** [comment]
> AFAIK the letters S and E are required for scrapers, TVDB

 but scrapers don't use the gui labels, hopefully?

**Comment by ksooo** [comment]
To sum up: No issue, as season number == 0 && episode number > 0 identifies a Special, which is correctly displayed as ESxx (where E stands for "Episode" and S stands for "Special".

**Comment by fuzzard** [comment]
No problem, thanks for looking into it.

**Comment by ksooo** [comment]
> This one should show S1E2 Demo TV Recording entry 11 Episode Name
> (screen shot shows ES2)

 Are you sure? pleaes double check.

**Comment by fuzzard** [comment]
Should the iSeriesNumber > 0 checks be removed from the channel, epg, timer checks in case LISTITEM_EPISODE: so they can be specials as well??

Dont worry about that one either ksooo, jalle19 pointed out an error (commented out a line that caused that)

**Comment by ksooo** [comment]
> Dont worry about that one either ksooo, jalle19 pointed out an error (commented out a line that caused that)

Yep, found it. https://github.com/kodi-pvr/pvr.demo/pull/58#discussion_r159822292

Hey, you're getting real time consuming. ;-) Anyway, keep up the good work.

**Comment by ksooo** [comment]
> Should the iSeriesNumber > 0 checks be removed from the channel, epg, timer checks in case LISTITEM_EPISODE: so they can be specials as well??

 Now I see! Thanks. Yes, this should be changed. Feel free to PR it.

**Comment by fuzzard** [comment]
I'll do a PR up tomorrow morning and send through.
